grudge/discretization.py

```diff
@@ -79,10 +84,10 @@ class DiscretizationCollection:
     """
 
     def __init__(self, array_context: ArrayContext, mesh: Mesh,
-                 order=None,
-                 discr_tag_to_group_factory=None, mpi_communicator=None,
-                 # FIXME: `quad_tag_to_group_factory` is deprecated
-                 quad_tag_to_group_factory=None):
+                 discr_tag_to_group_factory,
+                 discr_from_dd,
+                 dist_boundary_connections,
+                 mpi_communicator=None):
         """
```

Comment:
This version is definitely no longer compatible with code that calls the `DiscretizationCollection` constructor directly. That's OK for grudge because you've fixed all the call sites, and it's OK for mirgecom because it only uses `EagerDiscretization` to which you've added the backward compatibility code. But I'm not sure it's OK for all other users? (cc @alexfikl)

Alternatively, the compat code could live here.

Reply:
I have some grudge code sitting around, but it needs a lot of changes to move away from the deprecated symbolic stuff, so I'm fine with making a more breaking change here.

Reply:
Afraid I don't really know a way around this without resorting to some hacky python. I'd prefer not to make this overly complicated if possible. How about once this PR is merged, we pin an issue describing how the interface has changed?